### PR TITLE
8277397: ZGC: Add JFR event for temporary latency measurements

### PR DESCRIPTION
**Pull request by stefank**
I often measure latencies and stalls using JFR events. I'd like to add an event that can be used for these ad-hoc measurements during development and debugging.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277397](https://bugs.openjdk.java.net/browse/JDK-8277397): ZGC: Add JFR event for temporary latency measurements

### Reviewers
 * [Erik Österlund](https://openjdk.java.net/census#eosterlund) (@fisk - **Reviewer**)
 * [Jaroslav Bachorik](https://openjdk.java.net/census#jbachorik) (@jbachorik - **Reviewer**)
 * [Per Liden](https://openjdk.java.net/census#pliden) (@pliden - **Reviewer**)
 * [Markus Grönlund](https://openjdk.java.net/census#mgronlun) (@mgronlun - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6454/head:pull/6454` \
`$ git checkout pull/6454`

Update a local copy of the PR: \
`$ git checkout pull/6454` \
`$ git pull https://git.openjdk.java.net/jdk pull/6454/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6454`

View PR using the GUI difftool: \
`$ git pr show -t 6454`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6454.diff">https://git.openjdk.java.net/jdk/pull/6454.diff</a>

</details>

### CODE REVIEW

**Comment by stefank** [comment]
Retrying with correct labels.

**Review by fisk — APPROVED** [review]
Looks good.

File: src/jdk.jfr/share/conf/jfr/default.jfc [hunk 1]
```diff
@@ -775,6 +775,11 @@
       <setting name="threshold">0 ms</setting>
     </event>
 
+    <event name="jdk.ZThreadEvent">
```
Comment (jbachorik) [inline src/jdk.jfr/share/conf/jfr/default.jfc:778]: This is just a question - do we want/need to have this event enabled for the 'low overhead profiling' setup?

**Review by jbachorik — CHANGES_REQUESTED** [review]

File: src/hotspot/share/jfr/metadata/metadata.xml [hunk 2]
```diff
@@ -1034,6 +1034,11 @@
     <Field type="string" name="name" label="Name" />
   </Event>
 
+  <Event name="ZThreadEvent" category="Java Virtual Machine, GC, Detailed" label="ZGC Thread Event" thread="true" experimental="true">
```
[the remark this answers is not in the document]
Comment (stefank) [reply src/hotspot/share/jfr/metadata/metadata.xml:1037]: Great idea.

File: src/jdk.jfr/share/conf/jfr/default.jfc (hunk 1 above)
Comment (stefank) [reply src/jdk.jfr/share/conf/jfr/default.jfc:778]: Yes. I always use default.jfc. The profile.jfc is creating too many allocation related events to make it usable when running with ZGC. I know that there has been some plans to fix that, but I don't know if that made it into JDK 17 or not.

**Comment by stefank** [comment]
I discussed the naming of this event with @pliden, and we decided to rename it to make it more clear that this was only intended for debugging/development of ZGC.

**Review by pliden — APPROVED** [review]
Looks good.

**Review by fisk — APPROVED** [review]

**Review by jbachorik — APPROVED** [review]

**Comment by mgronlun** [comment]
What's the overhead of this event being enabled in default.jfc with threshold=0? Do all users really need to have this experimental event enabled by default when running ZGC?

**Comment by stefank** [comment]
> 

The overhead should be virtually zero, since no event is sent. Having this turned off by default makes this feature more annoying to use. Is this causing problems for the users?

**Comment by mgronlun** [comment]
Ah. So "debug" means only used in "debug builds" of the VM? In that case, there is no problem.

**Comment by stefank** [comment]
> 

No, this is more likely to be used in release builds to identify and measure performance issues in ZGC. It is intended for us during debugging and development of various features in ZGC.

**Comment by stefank** [comment]
Think of this similar to the UseNewCode flag.

**Review by mgronlun — APPROVED** [review]

**Pull request closed without merging**